### ISSUE TITLE
Fix lac/gmres_eigenvalues

### ISSUE THREAD

**Issue by tamiko** [dealii/dealii#6552]
In an attempt to make ICC happy, #6536 introduced a stopping criterion very close to floating point precision. This makes this test unstable. We should really fix this with something reliable. (That, for example, also survives more aggressive optimization.)

**Comment by kronbichler** [comment]
Since I've written the original test, I'd say that this test can indeed be relaxed. I used to make it similar to what some of the `gmres_reorthogonalize` tests (those are intrinsically roundoff tests and thus brittle) were doing because I wrote at about the same time, but it seems silly to try something similar here. We should just take a matrix that has a reasonable spread in eigenvalues and where we do not need more than say 20 iterations to make sure roundoff stays in check.
